packages/@aws-cdk/aws-route53/lib/record-set.ts

```diff
@@ -0,0 +1,440 @@
+
+  /**
+   * Explicity authorizes a single certificate authority to issue a
+   * wildcard certificate (and only wildcard) for the hostname.
+   */
+  ISSUEWILD = 'issuewild',
+
+  /**
+   * Specifies a URL to which a certificate authority may report policy
+   * violations.
+   */
+  IODEF = 'iodef',
+}
+
+/**
+ * Properties for a CAA record value.
+ */
+export interface CaaRecordValue {
+  /**
+   * The flag.
+   */
+  readonly flag: number;
+
+  /**
+   * The tag.
+   */
+  readonly tag: CaaTag;
+
+  /**
+   * The value associated with the tag.
+   */
+  readonly value: string;
+}
+
+/**
+ * Construction properties for a CaaRecord.
+ */
+export interface CaaRecordProps extends RecordSetOptions {
+  /**
+   * The values.
+   */
+  readonly values: CaaRecordValue[];
+}
+
+/**
+ * A DNS CAA record
+ *
+ * @resource AWS::Route53::RecordSet
+ */
+export class CaaRecord extends RecordSet {
+  constructor(scope: Construct, id: string, props: CaaRecordProps) {
+    super(scope, id, {
+      ...props,
+      recordType: RecordType.CAA,
+      recordValues: props.values.map(v => `${v.flag} ${v.tag} "${v.value}"`),
+    });
+  }
+}
+
+/**
+ * Construction properties for a CaaAmazonRecord.
+ */
+export interface CaaAmazonRecordProps extends RecordSetOptions {}
+
+/**
+ * A DNS Amazon CAA record.
+ *
+ * A CAA record to restrict certificate authorities allowed
+ * to issue certificates for a domain to Amazon only.
+ *
+ * @resource AWS::Route53::RecordSet
+ */
+export class CaaAmazonRecord extends CaaRecord {
+  constructor(scope: Construct, id: string, props: CaaAmazonRecordProps) {
+    super(scope, id, {
+      ...props,
+      values: [
+        {
+          flag: 0,
+          tag: CaaTag.ISSUE,
+          value: 'amazon.com'
+        }
+      ],
+      recordName: props.zone.zoneName
+    });
+  }
+}
+
+/**
+ * Properties for a MX record value.
+ */
+export interface MxRecordValue {
+  /**
+   * The priority.
+   */
+  readonly priority: number;
+
+  /**
+   * The mail server host name.
+   */
+  readonly hostName: string;
+}
+
+/**
+ * Construction properties for a MxRecord.
+ */
+export interface MxRecordProps extends RecordSetOptions {
+  /**
+   * The values.
+   */
+  readonly values: MxRecordValue[];
+}
+
+/**
+ * A DNS MX record
+ *
+ * @resource AWS::Route53::RecordSet
+ */
+export class MxRecord extends RecordSet {
+  constructor(scope: Construct, id: string, props: MxRecordProps) {
+    super(scope, id, {
+      ...props,
+      recordType: RecordType.MX,
+      recordValues: props.values.map(v => `${v.priority} ${v.hostName}`)
+    });
+  }
+}
+
+/**
+ * Construction properties for a ZoneDelegationRecord
+ */
+export interface ZoneDelegationRecordProps extends RecordSetOptions {
+  /**
+   * The name servers to report in the delegation records.
+   */
+  readonly nameServers: string[];
+}
+
+/**
+ * A record to delegate further lookups to a different set of name servers.
+ */
+export class ZoneDelegationRecord extends RecordSet {
```

Comment:
Shouldn't this be renamed to `NsRecord` for uniformity? (`ZoneDelegation` is the current name)